src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksTaxRuleDependencyServiceImpl.java

```diff
@@ -0,0 +1,261 @@
+    private static final Logger LOG = LogManager.getLogger(PaymentWorksTaxRuleDependencyServiceImpl.class);
+    
+    protected DateTimeService dateTimeService;
+    protected PaymentWorksBatchUtilityService paymentWorksBatchUtilityService;
+    protected ConfigurationService configurationService;
+    private DateTimeFormatter dateTimeFormatter;
+    
+
+    @Override
+    public KfsVendorDataWrapper populateTaxRuleDependentAttributes(PaymentWorksVendor pmwVendor,
+            Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap) {
+        LOG.info("populateTaxRuleDependentAttributes, entering");
+        KfsVendorDataWrapper vendorDataWrapper = new KfsVendorDataWrapper();
+        addVendorHeaderToKfsVendorDataWrapper(pmwVendor, paymentWorksIsoToFipsCountryMap, vendorDataWrapper);
+        return vendorDataWrapper;
+    }
+
+    protected void addVendorHeaderToKfsVendorDataWrapper(PaymentWorksVendor pmwVendor,
+            Map<String, List<PaymentWorksIsoFipsCountryItem>> paymentWorksIsoToFipsCountryMap, KfsVendorDataWrapper vendorDataWrapper) {
+        String vendorFipsCountryCode = convertIsoCountryCodeToFipsCountryCode(pmwVendor.getRequestingCompanyTaxCountry(), paymentWorksIsoToFipsCountryMap);
+        TaxRule taxRule = determineTaxRuleToUseForDataPopulation(pmwVendor, vendorFipsCountryCode);
+        
+        vendorDataWrapper.getVendorDetail().getVendorHeader().setVendorForeignIndicator(!isUnitedStatesFipsCountryCode(vendorFipsCountryCode));
+        
+        
+        LOG.info("addVendorHeaderToKfsVendorDataWrapper, vendor request " + pmwVendor.getPmwVendorRequestId() + " tax rule" + taxRule);
+        
+        if (taxRule.foreign) {
+            poulateForeignVendorValues(pmwVendor, vendorDataWrapper, taxRule);
+        } else {
+            vendorDataWrapper.getVendorDetail().getVendorHeader().setVendorTaxTypeCode(taxRule.taxTypeCode);
+            vendorDataWrapper.getVendorDetail().getVendorHeader().setVendorTaxNumber(pmwVendor.getRequestingCompanyTin());
+        }
+        
+        populateOwernshipCode(pmwVendor, taxRule, vendorDataWrapper.getVendorDetail().getVendorHeader());
+        vendorDataWrapper.getVendorDetail().getVendorHeader().setVendorCorpCitizenCode(convertIsoCountryCodeToFipsCountryCode(pmwVendor.getRequestingCompanyTaxCountry(), paymentWorksIsoToFipsCountryMap));
+        
+        if (taxRule.populateW9Attributes) {
+            populateW9Attributes(vendorDataWrapper, pmwVendor);
+        }
+        
+        if (taxRule.populateFirstLastLegalName) {
+            populateFirstLastLegalName(pmwVendor, vendorDataWrapper);
+        }
+        
+        if (taxRule.populateBusinessLegalName) {
+            populateBusinessLegalName(pmwVendor, vendorDataWrapper);
+        }
+    }
+    
+    protected TaxRule determineTaxRuleToUseForDataPopulation(PaymentWorksVendor pmwVendor, String pmwVendorFipsTaxCountryCode) {
+        TaxRule foundRule = TaxRule.OTHER;
```

Comment:
would this be better called just taxRule? foundRule denotes a boolean and this is not the case here.

Reply:
ok